### PR TITLE
Rectangular decomposition: Identity fix

### PR DESCRIPTION
**Pull request by shreyapkumar**
**Context:**
Currently, the rectangular decomposition is such that the identity matrix is implemented as a series of swaps. While this still gives the correct unitary, it is harder to implement on the hardware.

**Description of the Change:**
With this modification, the identity is decomposed such that there are no unnecessary swaps.

**Benefits:**
Easier to implement on hardware.

**Possible Drawbacks:**

**Related GitHub Issues:**
N/A

### CODE REVIEW

**Comment by josh146** [comment]
Thanks @shreyapkumar!

**Comment by josh146** [comment]
@lneuhaus it would be great to get your review on this, as the original decomposition author!

@shreyapkumar: can you also update the `.github/CHANGELOG.md` file, with a brief description of the change in the `Improvements` section, as well as a link back to this PR, and adding in your name in the contributors section 🙂

**Comment by lneuhaus** [comment]
> @lneuhaus it would be great to get your review on this, as the original decomposition author!

I only added the rectangular_symmetric depomposition, the other ones were written by @ishdhand I believe.

**Comment by josh146** [comment]
> I only added the rectangular_symmetric depomposition, the other ones were written by @ishdhand I believe.

Oops! Have added Ish. Feel free to review if you would like 🙂

File: strawberryfields/decompositions.py [hunk 1]
```diff
@@ -233,7 +233,10 @@ def nullTi(m, n, U):
     if nmax != mmax:
         raise ValueError("U must be a square matrix")
 
-    if U[m, n+1] == 0:
+    if (U[m, n+1] == 0 and U[m, n] == 0):
```
Comment (lneuhaus) [inline strawberryfields/decompositions.py]:
```suggestion
    if (U[m, n] == 0):
```
The proposed handling of this situation correct even if `U[m, n+1] != 0` and makes the code easier to read and saves a couple of function calls.

File: strawberryfields/decompositions.py [hunk 2]
```diff
@@ -251,7 +254,10 @@ def nullT(n, m, U):
     if nmax != mmax:
         raise ValueError("U must be a square matrix")
 
-    if U[n-1, m] == 0:
+    if (U[n-1, m] == 0 and U[n, m] == 0):
```
Comment (lneuhaus) [inline strawberryfields/decompositions.py]:
```suggestion
    if (U[n, m] == 0):
```
same comment as above

**Comment by josh146** [comment]
> In fact, looking at git blame, it seems like you implemented the original decompositions yourself in #7 @josh146 😄

🤦‍♂

**Comment by shreyapkumar** [comment]
Added the description to `CHANGELOG.md`. Thanks, @josh146  and @lneuhaus !

**Comment by co9olguy** [comment]
Was the updated changelog successfully committed/merged? I don't see it in this PR :thinking:

**Comment by co9olguy** [comment]
Answering my own question: looks like the changelog update was added in a separate PR (#313). 
@shreyapkumar For easier bookkeeping, and to save yourself and code reviewers extra effort,, simpler to keep changelog updates in same PR as the changes that are made :+1:

**Comment by shreyapkumar** [comment]
Thanks, will do that the next time, @co9olguy !  Wasn't able to figure out how to add to the same PR this time 🙈

**Pull request merged**